tsdb/chunks/head_chunks_test.go

```diff
@@ -297,42 +326,39 @@ func TestChunkDiskMapper_Truncate_PreservesFileSequence(t *testing.T) {
 	// though files 4 and 6 are empty.
 	file2Maxt := hrw.mmappedChunkFiles[2].maxt
 	require.NoError(t, hrw.Truncate(file2Maxt+1))
-	// As 6 was empty, it should not create another file.
```

Comment:
We need to test this. We should add a chunk and see that `3,4,5,6` is present and no `7`. And then truncate and add a chunk to create the file 7 (we need this truncate to make sure again that it preserves sequence of files).

Reply:
is this what you had in mind? 04e94b9df

Reply:
Not exactly, we should be truncating `file2Maxt+1` like before. Because the bug was the it was deleting empty files from later files.